lib/service/service.go

```diff
@@ -2568,11 +2568,14 @@ func (process *TeleportProcess) initUploaderService() error {
 		return trace.Wrap(err)
 	}
 
-	log.Infof("starting upload completer service")
-
 	connectors := process.getConnectors()
 	var conn *Connector
 	for _, c := range connectors {
+		// Skip types.RoleMDM, MDM services don't have the necessary permissions to
+		// run the uploader.
+		if c.ServerIdentity != nil && c.ServerIdentity.ID.Role == types.RoleMDM {
```

Comment:
Do we want to use `ServerIdentity` instead of `ClientIdentity` ? According to the comments the `ClientIdentity` is used  "as internal cluster clients to the auth service."

>> 	// ClientIdentity is the identity to be used in internal cluster
	// clients to the auth service.
	ClientIdentity *auth.Identity

>> 	// ServerIdentity is the identity to be used in servers - serving SSH
	// and x509 certificates to clients.
	ServerIdentity *auth.Identity

Reply:
Well spotted, changed the log statement below too.